### PR TITLE
Updated IoT Central .NET package to 4.0.0 (RP version 2021-06-01) with support for Managed Identities

### PR DESCRIPTION
**Pull request by gansvv**
## Description

Updated IoT Central Management Plane .Net package to 4.0.0, which references IoT Central's RP release v2021-06-01. 

This update adds support for Managed Identities, removes support for S1 SKU, and allows calls using regional locations (migrating off geographic). Also updated models and tests for the new features.

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/IotCentral/IotCentral/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Upcoming Release
 
+## Version 1.0.0
+* Stable release version, updated to IoT Central .NET Management Plane SDK version 4.0.0. 
+* This change introduces support for System-Assigned Managed Identities, adds support for regional locations like 'eastus' (while removing support for geographic locations like 'unitedstates'), and removes support for legacy S1 sku.
```
Comment (r-delgadillo) [inline src/IotCentral/IotCentral/ChangeLog.md:24]: NIT: Support for regional locations is existing behavior in api version 2018-09-01, the only difference is deprecating support for geographical resources.

**Review by r-delgadillo — DISMISSED** [review]

File: src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs [hunk 2]
```diff
@@ -67,7 +72,23 @@ public override void ExecuteCmdlet()
...
         {
-            return new AppPatch(TagsConversionHelper.CreateTagDictionary(this.Tag, true), new AppSkuInfo(this.Sku), null, this.DisplayName, this.Subdomain);
+            var appPatch = new AppPatch(
+                tags: TagsConversionHelper.CreateTagDictionary(this.Tag, true),
+                sku: new AppSkuInfo(this.Sku),
+                displayName: this.DisplayName,
+                subdomain: this.Subdomain
+            );
+
+            if (!string.IsNullOrEmpty(this.Identity))
```
[the remark this answers is not in the document]
Comment (poanchen) [reply src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs:82]: probably just check if this.Identity has value
if it does, go ahead and change it
if it doesn't, do nothing

**Review by poanchen — APPROVED** [review]

File: src/IotCentral/IotCentral/ChangeLog.md (hunk 1 above)
Comment (gansvv) [reply src/IotCentral/IotCentral/ChangeLog.md:24]: Updated changelog, thanks.

File: src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md [hunk 3]
```diff
@@ -14,21 +14,21 @@ Updates the metadata for an IoT Central Application.
...
-Set-AzIotCentralApp [-DisplayName <String>] [-Subdomain <String>] [-Tag <Hashtable>] [-Sku <String>]
+Set-AzIotCentralApp [-DisplayName <String>] [-Subdomain <String>] [-Tag <Hashtable>] [-Sku <String>] [-Identity <String>]
  -InputObject <PSIotCentralApp> [-AsJob] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
  [<CommonParameters>]
 ```
 
 ### InteractiveIotCentralParameterSet
 ```
-Set-AzIotCentralApp [-DisplayName <String>] [-Subdomain <String>] [-Tag <Hashtable>] [-Sku <String>] [-AsJob]
+Set-AzIotCentralApp [-DisplayName <String>] [-Subdomain <String>] [-Tag <Hashtable>] [-Sku <String>] [-Identity <String>] [-AsJob]
```
[the remark this answers is not in the document]
Comment (gansvv) [reply src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md:31]: There is a separate model for user assigned identities. Per the swagger feedback from design team, we should use the MSI (Managed System Identity) model which supports both System and User MIs.

File: src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gansvv) [reply src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs:82]: Thanks, I've removed the else and added validation on tests.

**Review by poanchen — DISMISSED** [review]
LGTM

**Comment by gansvv** [comment]
@isra-fel Please review and help merge it back into Azure repo.

**Review by wyunchi-ms — APPROVED** [review]

**Review by poanchen — APPROVED** [review]

**Comment by wyunchi-ms** [comment]
Looks you want to GA this module. Please follow the process of our [guideline](https://eng.ms/docs/cloud-ai-platform/azure/azure-core-compute/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azps_docs/process_ga_module).

File: src/IotCentral/IotCentral/Az.IotCentral.psd1 [hunk 4]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.9.0'
+ModuleVersion = '1.0.0'
```
Comment (isra-fel) [inline src/IotCentral/IotCentral/Az.IotCentral.psd1:15]: - Module version is calculated, so please revert this change
- `1.0.0` means this powershell module is GA. GA is a big deal so if this is not a mistake, please refer to [Process of GA](https://github.com/Azure/azure-powershell-pr/wiki/Process-of-GA-module)

File: src/IotCentral/IotCentral/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -32,5 +32,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("0.7.4")]
-[assembly: AssemblyVersion("0.9.0")]
-[assembly: AssemblyFileVersion("0.9.0")]
+[assembly: AssemblyVersion("1.0.0")]
+[assembly: AssemblyFileVersion("1.0.0")]
```
Comment (isra-fel) [inline src/IotCentral/IotCentral/Properties/AssemblyInfo.cs:36]: Versions in this file are also generated. Please revert it

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/IotCentral/IotCentral/Properties/AssemblyInfo.cs (hunk 5 above)
Comment (gansvv) [reply src/IotCentral/IotCentral/Properties/AssemblyInfo.cs:36]: Done.

File: src/IotCentral/IotCentral/Az.IotCentral.psd1 (hunk 4 above)
Comment (gansvv) [reply src/IotCentral/IotCentral/Az.IotCentral.psd1:15]: Done

**Review by isra-fel — APPROVED** [review]
Looks good to me :)
